### PR TITLE
Move e2e tests to independent folder

### LINKED ISSUE #219: Move all end-to-end tests to special folder

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
Closes #219

### CODE REVIEW

**Review by mpoke — CHANGES_REQUESTED** [review]
Nice work @smarshall-spitzbart. I didn't check the functionality, since I guess it's just a refactoring, right? 

At the moment the structure is a bit hard to follow, maybe a README.md would help, i.e., 
- `setup_test.go` -- setup for the e2e tests
- `common_test.go` -- helper functions
- `channel_init_test.go` -- e2e tests for the _Channel Initialization_ sub-protocol
- `valset_update_test.go` -- e2e tests for the _Validator Set Update_ sub-protocol
- `unbonding_test.go` -- e2e tests for the _Completion of Unbonding Operations_
- `slashing_test.go` -- e2e tests for the _Consumer Initiated Slashing_ sub-protocol
- `distribution_test.go` -- e2e tests for the _Reward Distribution_ sub-protocol
- `stop_consumer_test.go` -- e2e tests for the _Consumer Chain Removal_ sub-protocol

Move to `valset_update.go` the content from `bonding_test.go` and `vsc_test.go`. 

Move to `channel_init_test.go` the content from `genesis_test.go`, `ibc_test.go`, `history_test.go`, `proposal_test.go`.

`distribution_test.go` will be empty for now. 

Re. the 4 different `...TestSuite`s data structure, I think we can have only one, but let's do that in a separate PR.

**Comment by smarshall-spitzbart** [comment]
> Nice work @smarshall-spitzbart. I didn't check the functionality, since I guess it's just a refactoring, right?
> 
> At the moment the structure is a bit hard to follow, maybe a README.md would help, i.e.,
> 
> * `setup_test.go` -- setup for the e2e tests
> * `common_test.go` -- helper functions
> * `channel_init_test.go` -- e2e tests for the _Channel Initialization_ sub-protocol
> * `valset_update_test.go` -- e2e tests for the _Validator Set Update_ sub-protocol
> * `unbonding_test.go` -- e2e tests for the _Completion of Unbonding Operations_
> * `slashing_test.go` -- e2e tests for the _Consumer Initiated Slashing_ sub-protocol
> * `distribution_test.go` -- e2e tests for the _Reward Distribution_ sub-protocol
> * `stop_consumer_test.go` -- e2e tests for the _Consumer Chain Removal_ sub-protocol
> 
> Move to `valset_update.go` the content from `bonding_test.go` and `vsc_test.go`.
> 
> Move to `channel_init_test.go` the content from `genesis_test.go`, `ibc_test.go`, `history_test.go`, `proposal_test.go`.
> 
> `distribution_test.go` will be empty for now.
> 
> Re. the 4 different `...TestSuite`s data structure, I think we can have only one, but let's do that in a separate PR.

You're correct that I didn't change any test functionality in this PR, I only moved things around. 

Your mentioned file organization sounds great to me, thank you! My only correction to your proposed solution is that I believe ```TestTrackHistoricalInfo``` should go in its own file under ```normal_operations_test.go```, since it doesn't relate to channel initialization. A new readme was added to the e2e test directory.

Lastly I've opened #300 to address the fact that we have a lot of redundant code in ```setup_test.go```.

**Review by mpoke — APPROVED** [review]
Great work.

File: e2e-tests/valset_update_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	appConsumer "github.com/cosmos/interchain-security/app/consumer"
+	consumertypes "github.com/cosmos/interchain-security/x/ccv/consumer/types"
+	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
+	"github.com/cosmos/interchain-security/x/ccv/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// TestPacketRoundtrip tests a CCV packet roundtrip when tokens are bonded on provider
+func (s *ProviderTestSuite) TestPacketRoundtrip() {
```
Comment (jtremback) [inline e2e-tests/valset_update_test.go:18]: maybe put this in normal_operations_test

File: .github/workflows/automated-tests.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Figure out where we are
         run: ls
```
Comment (jtremback) [inline .github/workflows/automated-tests.yml:15]: Pretty sure this can be removed and i left it in here from debugging

**Review by jtremback — APPROVED** [review]
